docs/basic-features/data-fetching.md

```diff
@@ -4,94 +4,392 @@ description: Next.js can handle data fetching in multiple ways for server-render
+  return {
+    props: {
+      posts,
+    },
+  }
 }
 
-export default HomePage
+export default Blog
 ```
 
-## Server-side rendering
+### When should I use `getStaticProps`?
+
+You should use `getStaticProps` if:
+
+- The data required to render the page is available at build time ahead of a user's request.
+- The data comes from headless CMS.
+- The data can be publicly cached (not user-specific).
+- The page must be pre-rendered (for SEO) and be very fast — `getStaticProps` generates HTML and JSON files, both of which can be cached by a CDN for performance.
+
+### Technical details
+
+#### Only runs at build time
+
+Because `getStaticProps` runs at build time, it does **not** receive data that's only available during request time, such as query parameters or HTTP headers as it generates static HTML.
+
+#### Write server-side code directly
+
+Note that `getStaticProps` runs only on the server-side. It will never be run on the client-side. It won't even be included in the JS bundle for the browser. That means you can write code such as direct database queries without them being sent to browsers. You should not fetch an **API route** from `getStaticProps` — instead, you can write the server-side code directly in `getStaticProps`.
+
+#### Statically Generates both HTML and JSON
+
+When a page with `getStaticProps` is pre-rendered at build time, in addition to the page HTML file, Next.js generates a JSON file holding the result of running `getStaticProps`.
+
+This JSON file will be used in client-side routing through `next/link` ([documentation](/docs/api-reference/next/link.md)) or `next/router` ([documentation](/docs/api-reference/next/router.md)). When you navigate to a page that's pre-rendered using `getStaticProps`, Next.js fetches this JSON file (pre-computed at build time) and uses it as the props for the page component. This means that client-side page transitions will **not** call `getStaticProps` as only the exported JSON is used.
 
-Pages might have to wait on some data before pre-rendering, for example, if you want the page to be indexed with the data by search engines.
+#### Only allowed in a page
 
-Next.js comes with [`getInitialProps`](/docs/api-reference/data-fetching/getInitialProps.md), which is an [`async`](https://zeit.co/blog/async-and-await) function that can be added to any page as a [`static method`](https://javascript.info/static-properties-methods).
+`getStaticProps` can only be exported from a **page**. You can't export it from non-page files.
 
-`getInitialProps` allows the page to wait for data before rendering starts.
+One of the reasons for this restriction is that React needs to have all the required data before the page is rendered.
 
-Using `getInitialProps` will make the page opt-in to on-demand [server-side rendering](/docs/basic-features/pages.md#server-side-rendering).
+Also, you must use `export` — it will **not** work if you add `getStaticProps` as a property of the page component.
```

Comment:
I might be daft, but the first time I read this I thought I had to use `next export`.

Reply:
Updated!